packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,155 @@
-   *
-   * @returns {Promise.<string>} the token
-   */
-  async retrieveEDLToken(): Promise<string> {
-    const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
-    const url = this.getEDLurl();
-    // response: get a token from the Earthdata login endpoint using credentials if exists
-    let response: EarthdataGetTokenResponse;
-    try {
-      response = await got.get(`${url}/api/users/tokens`,
-        {
-          headers: {
-            Authorization: `Basic ${buff}`,
-          },
-        }).json();
-    } catch (error) {
-      this.handleHttpError(error);
-    }
-    const currDate = new Date();
-
-    for (let i = 0; i < Object.keys(response).length; i += 1) {
-      const responseDate = new Date(response[i].expiration_date);
-      if (currDate < responseDate) {
-        return response[i].access_token;
-      }
-    }
-    return undefined;
+/**
+  * The method for creating Earthdata Login token. This method sends a POST request
+  * to the Earthdata Login endpoint URL in order to create a token for the user. The users'
+  * username and password are sent as base64 encoded credentials as a header for
+  * authorization. If the users' credentials are accepted a token is created on their
+  * behalf and returned, if not, an error is thrown.
+  *
+  * @returns {string} the token
+*/
+export const createEDLToken = async (username: string, password: string, edlEnv: string):
+Promise<string> => {
+  let rawResponse: any;
+  try {
+    rawResponse = await got.post(`${getEdlUrl(edlEnv)}/api/users/token`,
+      {
+        responseType: 'json',
+        username: username,
+        password: password,
+      });
+  } catch (error) {
+    if (error instanceof got.HTTPError) throw parseHttpError(error, 'create');
+    throw parseCaughtError(error);
   }
+  const response = PostTokenResponseBody.parse(rawResponse.body);
+  return response[0].access_token;
+};
 
-  /**
-   * The method for creating Earthdata Login token. This method sends a POST request
-   * to the Earthdata Login endpoint URL in order to create a token for the user. The users'
-   * username and password are sent as base64 encoded credentials as a header for
-   * authorization. If the users' credentials are accepted a token is created on their
-   * behalf and returned, if not, an error is thrown.
-   *
-   * @returns {Promise.<string>} the token
-   */
-  async createEDLToken(): Promise<string> {
-    const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
-    const url = this.getEDLurl();
-    let response: EarthdataPostTokenResponse;
-    try {
-      response = await got.post(`${url}/api/users/token`,
-        {
-          headers: {
-            Authorization: `Basic ${buff}`,
-          },
-        }).json();
-    } catch (error) {
-      this.handleHttpError(error);
-    }
-    return response[0].access_token;
+/**
+  * This method is used for the cmrTokenSpec integration test in order to revoke the
```

Comment:
The indentation of these JSDocs needs to be fixed. The `*` here should be under the first `*` above.